File: TMTDentalAPI/ClientApp/src/app/products/product-step.ts
import { Product } from './product';

export class ProductStepDisplay {
    id: string;
    name: string;
    order: number;
    note: string;
    productId: string;
    product: Product;
    default: boolean;
}

export class ProductStepSimple {
    id: string;
    name: string;
}

export class ProductDisplayAndStep extends Product {
    stepList: ProductStepDisplay[];
}

File: ApplicationCore/Entities/ProductStep.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class ProductStep : BaseEntity
    {
        public ProductStep()
        {
            Active = true;
            Default = true;
        }
        /// <summary>
        /// Tên 
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Thứ tự
        /// </summary>
        public int Order { get; set; }

        /// <summary>
        /// Ghi chú
        /// </summary>
        public string Note { get; set; }

        /// <summary>
        /// Tham chiếu Product.type Service
        /// </summary>
        public Guid? ProductId { get; set; }
        public Product Product { get; set; }

        /// <summary>
        /// Trạng thái
        /// </summary>
        public bool Active { get; set; }

        /// <summary>
        /// Mặc định 
        /// </summary>
        public bool Default { get; set; }

    }
}
